### ISSUE TITLE
Conan Package

### ISSUE THREAD

**Issue by ChristianPanov** [ChristianPanov/lwlog#39]

**Comment by autoantwort** [comment]
I would do it. Is there a reason why this lib can not be a shared lib?
https://github.com/ChristianPanov/lwlog/blob/a01cc8da67d76ebd73bdf5e1d18fa46a99a60420/CMakeLists.txt#L23

**Comment by ChristianPanov** [comment]
Refer to this comment regarding your question: https://github.com/ChristianPanov/lwlog/pull/49#issuecomment-2569783418

In any case, thank you for the support. I've tried setting up both Conan and VCPKG, but there are some problems.
The Conan PR is still stuck in the CI - https://github.com/conan-io/conan-center-index/pull/26283.
And the VCPKG PR is also stuck in CI, and although someone was assigned to the issue, there is not much progress - https://github.com/microsoft/vcpkg/pull/43047

**Comment by autoantwort** [comment]
The ci fails because the version database contains the wrong entry

**Comment by ChristianPanov** [comment]
Yes, I noticed that the versioning job only fails for the x86 workflow, could it be because 32 bit architecture is not supported? Although on a second thought, a lot of the jobs in other workflows are skipped, probably due to this error, so they might not be successful as well.

**Comment by autoantwort** [comment]
The version database is only checked for one arch since it would be the same check for every workflows. 
```
D:\a\_work\1\s\versions\l-\lwlog.json: error: lwlog@1.3.0 git tree 03bafbae9158d058b2c5d7cd5eb13ca28b8f30da does not match the port directory
D:\a\_work\1\s\ports\lwlog: note: the port directory has git tree 52f20d71e449b8edb6223e8c4316ea06c5ce6015
D:\a\_work\1\s\ports\lwlog\vcpkg.json: note: if lwlog@1.3.0 is already published, update this file with a new version or port-version, commit it, then add the new version by running:
  vcpkg x-add-version lwlog
  git add versions
  git commit -m "Update version database"
note: if lwlog@1.3.0 is not yet published, overwrite the previous git tree by running:
  vcpkg x-add-version lwlog --overwrite-version
  git add versions
  git commit -m "Update version database"
```

**Comment by ChristianPanov** [comment]
I see. I will wait till tuesday to see if anything will be resolved; if not, I will assign the issue to you if you still want to do it

**Comment by autoantwort** [comment]
Ah I only wanted to do the vcpkg port 😅

**Comment by neeraj9** [comment]
@ChristianPanov, Thanks for this project and pushing to vcpkg. A minot nitpick though is that vcpkg installs the include for lwlog to `include/src` instead of `include/lwlog`. So in my code I have to `#include <src/lwlog.h>` which is bad.

My `build/vcpkg_installed/vcpkg/info/lwlog_1.3.0_x64-windows.list` looks like this for your reference.

```
x64-windows/
x64-windows/debug/
x64-windows/debug/lib/
x64-windows/debug/lib/lwlog_lib.lib
x64-windows/include/
x64-windows/include/src/
x64-windows/include/src/configuration.h
x64-windows/include/src/details/
x64-windows/include/src/details/argument_format.h
x64-windows/include/src/details/argument_format_impl.h
x64-windows/include/src/details/bounded_queue.h
x64-windows/include/src/details/bounded_queue_impl.h
x64-windows/include/src/details/log_message.h
...
```

**Comment by ChristianPanov** [comment]
> 
> @ChristianPanov, Thanks for this project and pushing to vcpkg. A minot nitpick though is that vcpkg installs the include for lwlog to `include/src` instead of `include/lwlog`. So in my code I have to `#include <src/lwlog.h>` which is bad.
> 
> My `build/vcpkg_installed/vcpkg/info/lwlog_1.3.0_x64-windows.list` looks like this for your reference.
> 
> ```
> x64-windows/
> x64-windows/debug/
> x64-windows/debug/lib/
> x64-windows/debug/lib/lwlog_lib.lib
> x64-windows/include/
> x64-windows/include/src/
> x64-windows/include/src/configuration.h
> x64-windows/include/src/details/
> x64-windows/include/src/details/argument_format.h
> x64-windows/include/src/details/argument_format_impl.h
> x64-windows/include/src/details/bounded_queue.h
> x64-windows/include/src/details/bounded_queue_impl.h
> x64-windows/include/src/details/log_message.h
> ...
> ```

@neeraj9 Thank you for pointing that out, I was planning to have the directory fixed in the next release, it is indeed bad, it forces very awkward file paths.

 Do you mind opening an issue about it actually?

Thanks

**Comment by neeraj9** [comment]
Done - https://github.com/ChristianPanov/lwlog/issues/56

**Comment by ChristianPanov** [comment]
> Done - https://github.com/ChristianPanov/lwlog/issues/56

Thank you very much
